File: backend/app/modules/student/schemas/student.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class StudentDiplomaListItem(BaseModel):
  id: UUID
  verification_hash: str
  graduation_year: int
  status: Literal["valid", "annulled"]


class StudentDiplomaListResponse(BaseModel):
  items: list[StudentDiplomaListItem]
  total: int
  limit: int
  offset: int


class StudentDiplomaDetailResponse(BaseModel):
  id: UUID
  verification_hash: str
  graduation_year: int
  status: Literal["valid", "annulled"]
  full_name: str
  specialty: str
  diploma_number: str


class CreateAccessTokenRequest(BaseModel):
  diploma_id: UUID
  ttl_days: int = Field(default=7, ge=1, le=90)


class AccessTokenResponse(BaseModel):
  id: UUID
  diploma_id: UUID
  token_value: str
  share_url: str
  expires_at: datetime
  is_revoked: bool
  created_at: datetime


class AccessTokenListItem(BaseModel):
  id: UUID
  token_value: str
  expires_at: datetime
  is_revoked: bool
  created_at: datetime


class AccessTokenListResponse(BaseModel):
  items: list[AccessTokenListItem]
  total: int
  limit: int
  offset: int


class RevokeAccessTokenResponse(BaseModel):
  id: UUID
  is_revoked: bool

File: frontend/src/shared/types/student.ts
export const studentDiplomaStatuses = ['valid', 'annulled'] as const
export const studentAccessTokenStatuses = ['active', 'revoked', 'expired'] as const

export type StudentDiplomaStatus = (typeof studentDiplomaStatuses)[number]
export type StudentAccessTokenStatus = (typeof studentAccessTokenStatuses)[number]

export interface StudentDiplomaListItem {
  id: string
  verification_hash: string
  graduation_year: number
  status: StudentDiplomaStatus
}

export interface StudentDiplomaListResponse {
  items: StudentDiplomaListItem[]
  total: number
  limit: number
  offset: number
}

export interface StudentDiplomaDetail {
  id: string
  verification_hash: string
  graduation_year: number
  status: StudentDiplomaStatus
  full_name: string
  specialty: string
  diploma_number: string
}

export interface StudentAccessTokenListItem {
  id: string
  token_value: string
  share_url: string
  expires_at: string
  created_at: string
  is_revoked: boolean
  status: StudentAccessTokenStatus
}

export interface StudentAccessTokenListResponse {
  items: StudentAccessTokenListItem[]
  total: number
  limit: number
  offset: number
}

export interface StudentAccessTokenResponse {
  id: string
  diploma_id: string
  token_value: string
  share_url: string
  expires_at: string
  created_at: string
  is_revoked: boolean
  status: StudentAccessTokenStatus
}

export interface CreateStudentAccessTokenPayload {
  diploma_id: string
  ttl_days?: number
}
